src/components/HotspotListItem.tsx

```diff
@@ -25,69 +24,74 @@ const HotspotListItem = ({
   showCarot = false,
 }: HotspotListItemProps) => {
   const { t } = useTranslation()
-
+  const { toggleConvertHntToCurrency } = useCurrency()
+  const handlePress = useCallback(() => onPress?.(hotspot), [hotspot, onPress])
```

Comment:
Can we add a debounce to this? Seems to take some time so spamming it might be bad